encoding/thrift/thriftrw-plugin-yarpc/main.go

```diff
@@ -93,6 +79,24 @@ func splitFunctionPath(input string) (string, string) {
 	return input[:i], input[i+1:]
 }
 
+func buildSvc(serviceID api.ServiceID, req *api.GenerateServiceRequest) *Svc {
+	service := req.Services[serviceID]
+	module := req.Modules[service.ModuleID]
+
+	var parents []*Svc
+	if service.ParentID != nil {
+		parentSvc := buildSvc(*service.ParentID, req)
+		parents = append(parents, parentSvc)
+		parents = append(parents, parentSvc.Parents...)
```

Comment:
Thrift supports single inheritance but the parent can inherit another service
and so on. Parents is a list of the parent service stack. It's like "mro" in
Python.